### ISSUE TITLE
Paperless, but without GPG

### ISSUE THREAD

**Issue by guillaumededrie** [danielquinn/paperless#188]
Hi,
@danielquinn, thank you very much for this useful project!

I just pushed a version of `Paperless` but without GPG. You can take a look at [the code](https://github.com/guillaumededrie/paperless-without-gpg).

I am not a really big fan of putting a GPG passphrase in a config file (mainly for security reason), so instead, I set up an encrypted directory on my NAS (key on RAM, so cold boot attack are possible) and Paperless do the rest! (I put a [readme](https://github.com/guillaumededrie/paperless-without-gpg) to explain in detail my setup).

At first it was just for me, but I realized that maybe some people can be interested by. 

I don't know if we can make an option (i.e. the user can choose not to encrypt PDFs), push the code on a different branch, or simply add a link somewhere in the README/Documentation.

Thank you again,

-- 
Guillaume

**Comment by danielquinn** [comment]
Well the changes you've made there are rather drastic, so I honestly wouldn't consider supporting both methods -- the effort involved is just too much.  The project is GPL though, so if you wanna fork it for this purpose, go nuts :-)

A few suggestions though:

* Your changes include editing migrations which is a big no-no.  That's historical data for Paperless users, so making those sorts of changes would effectively make it impossible for someone to easily switch to an unencrypted setup like yours.  Either you start with an empty `migrations` directory, or you leave it alone and write a migration to decrypt your user's data so they can use your setup.
* Storing a password in a config file is definitely not optimal, but you'll note that [in the manage.py file there's an option to specify it at start time](https://github.com/danielquinn/paperless/blob/master/src/manage.py#L16) thereby saving you the dangers of storing the value in a config.

The truth is, I don't much like the use of GPG here, and in retrospect, I would have preferred to go with Python's excellent [cryptograpy](https://pypi.python.org/pypi/cryptography) module instead.  The real trouble spot however is that Paperless is expected not just to *store* sensitive data but to *serve it back up* on request, so whichever path I chose it had to be symmetric encryption unless we get into complex stuff like storing a private key in a cookie on each user's computer and... well that's a mess for a self-hosted project like this.

So to sum up, I'm glad you're finding Paperless so useful and that you were able to modify it for your own purposes.  Supporting a fork like yours though, which such drastic differences, isn't something I'm keen on doing.

**Comment by Gummibeer** [comment]
I struggled over the same thing.
Until now I have a Synology with a `documents` drive - this is organized in folders and without search - so the oldschool bad way. At the weekend I've tested paperless and it works pretty good - even on my RasPi! :)
But to get it working I've still my `documents` drive with all the files unencrypted and another `paperless` drive with an `import` and a `storage` folder - with a simple script I copy all files from the `documents` drive to the `paperless/import` folder but this way I double the needed storage and it's also pretty complicated. :/

So two things that would help/solve this issue:
* keep the files in the consumption directory (saves the copy'n'paste script)
* disable encryption - saves me the whole `paperless` drive

Is there any plan to add one of these features?

**Comment by danielquinn** [comment]
Not at the moment, no.  Keeping files in the consumption directory would mean that the consumer would attempt to re-import everything on server start.  Not awesome.

Disabling encryption by default won't happen, though I'm open to making it configurable.  Something like a flag in [this file](https://github.com/danielquinn/paperless/blob/master/src/paperless/db.py) that would magically switch encryption on or off, coupled with documentation that explains that this is a one-time decision thing would be ok, but it's not a priority for me right now.  If someone were to issue a PR for that though, I'd work with them to make it fit.

@guillaumededrie's fork is just too drastic to consider, but a flag is doable should someone feel the need to do the work.

**Comment by d4rken** [comment]
Any update on making encryption optional via flag? A flag in the linked file that just copies the files would probably work? Though file naming should also not use `.gpg` if encryption is disabled...

**Comment by danielquinn** [comment]
The update is that I have no plans to implement this in a near future.  I've got too much on my plate right now and disabling encryption just isn't a priority for me.

**Comment by d4rken** [comment]
@danielquinn I understand completely.

@guillaumededrie Do you plan to adopt your changes (like @danielquinn mentioned) so they can be merged back? 

@Gummibeer Did you make any progress on this?

**Comment by Gummibeer** [comment]
@d4rken my progress is out of this project. python and primary django aren't my favorite tools, there are some more things I want another way (filenames, storage engines etc). So I developed my own - atm not public.
